File: xLiAd.DiagnosticLogCenter.Agent/FilterRule.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.RegularExpressions;
using System.Xml.XPath;

namespace xLiAd.DiagnosticLogCenter.Agent
{
    public static class FilterRule
    {
        internal static IEnumerable<string> AllowPath { get; set; }
        internal static IEnumerable<string> ForbbidenPath { get; set; }

        internal static bool ShouldRecord(string path)
        {
            if (AllowPath.AnyX())
            {
                if (path.NullOrEmpty())
                    return false;
                foreach (var item in AllowPath)
                {
                    bool r = Regex.IsMatch(path, item);
                    if (r)
                        return true;
                }
                return false;
            }
            else if (ForbbidenPath.AnyX())
            {
                if (path.NullOrEmpty())
                    return true;
                foreach (var item in AllowPath)
                {
                    bool r = Regex.IsMatch(path, item);
                    if (r)
                        return false;
                }
                return true;
            }
            else
                return true;
        }
    }
}

File: DiagnosticLogCenter-Java/src/main/java/com/xliad/diagnosticlogcenter/agent/FilterRule.java
package com.xliad.diagnosticlogcenter.agent;

import java.util.List;
import java.util.regex.Pattern;

public class FilterRule {
    private static List<String> allowPath;
    private static List<String> forbiddenPath;

    public static void setAllowPath(List<String> paths) {
        allowPath = paths;
    }

    public static void setForbiddenPath(List<String> paths) {
        forbiddenPath = paths;
    }

    public static boolean shouldRecord(String path) {
        System.out.println("FilterRule.shouldRecord: " + path);
        if (path == null) {
            path = "";
        }

        if (allowPath != null && !allowPath.isEmpty()) {
            for (String pattern : allowPath) {
                if (Pattern.matches(pattern, path)) {
                    return true;
                }
            }
            return false;
        } else if (forbiddenPath != null && !forbiddenPath.isEmpty()) {
            for (String pattern : forbiddenPath) {
                if (Pattern.matches(pattern, path)) {
                    return false;
                }
            }
            return true;
        } else {
            return true;
        }
    }
}